src/libutil/util.hh

```diff
@@ -62,7 +62,7 @@ Path canonPath(PathView path, bool resolveSymlinks = false);
    everything before the final `/'.  If the path is the root or an
    immediate child thereof (e.g., `/foo'), this means `/'
    is returned.*/
-Path dirOf(const Path & path);
+PathView dirOf(const PathView path);
```

Comment:
In my experience returning string_views from functions is a great way to introduce subtle bugs when the lifetime of the returned value exceeds the lifetime of the argument... It's also not nice that you can't pass it temporary values.

Reply:
if only C++ had lifetimes, then this wouldn't be that much of an issue :disappointed: what's the best way forward here then? change, revert, something else? `dirOf` in particular isn't very hot and most of its users do make a copy of its return value, so undoing that wouldn't hurt much.

Reply:
Yeah, unless it's on a hot path it's probably not worth it. I have bad experience writing things like `auto s = dirOf(bar())` where `bar()` returns a temporary...